### ISSUE TITLE
Add Boa to Test262-Harness

### ISSUE THREAD

**Issue by jasonwilliams** [boa-dev/boa#3431]
Boa being part of [Test262-Harness](https://github.com/bterlson/test262-harness#test262-harness) will make it easier to test which parts of TEST262 Boa has implemented. Currently it's not easy to do this. There are a few dependencies Boa needs to be added to before this can happen.

Test262-harness uses eshost which uses esvu. Boa support would need to be added to these repositories.

- [x] https://github.com/devsnek/esvu/pull/53
- [ ] https://github.com/bterlson/eshost
- [ ] https://github.com/bterlson/test262-harness

**Comment by linusg** [comment]
Upstream eshost and test262-harness are effectively unmaintained, so test262.fyi uses these two forks:

- https://github.com/CanadaHonk/eshost
- https://github.com/CanadaHonk/test262-harness

@CanadaHonk [added boa support](https://github.com/CanadaHonk/eshost/blob/master/runtimes/boa.js) to the eshost fork, test262-harness doesn't need any additional changes.

**Comment by jasonwilliams** [comment]
This is good to know and thanks for adding Boa to these forks.

My concern though is that the TC39 champions are still using the original version. Last time I spoke to Temporal champions they were using [Brian’s harness](https://github.com/bterlson/test262-harness), it’s also listed here on the test262 page. https://github.com/tc39/test262?tab=readme-ov-file#test262-runners

Is it possible to make @canadaHonk’s version the canonical version by asking on there? Or I can ask if you don’t want to. We should just have everyone use this one if it’s more actively maintained

**Comment by CanadaHonk** [comment]
Fwiw, I haven't fully tried to get my changes upstream as those were originally mostly intended as a fork for test262.fyi specifically before I knew upstream was inactive. I believe they are still usable standalone like the original though, so I'm happy for my forks to become canonical if wanted.

**Comment by jasonwilliams** [comment]
For what it’s worth, I don’t think it makes sense to even try upstreaming your changes at this point. The upstream version hasn’t been maintained in years and Brian isn’t as active as he was before either.

Effort is better spent moving to your version (updating docs etc), but that’s just my opinion.

**Comment by jasonwilliams** [comment]
> Upstream eshost and test262-harness are effectively unmaintained, so test262.fyi uses these two forks:
> 
> * https://github.com/CanadaHonk/eshost
> * https://github.com/CanadaHonk/test262-harness
> 
> @CanadaHonk [added boa support](https://github.com/CanadaHonk/eshost/blob/master/runtimes/boa.js) to the eshost fork, test262-harness doesn't need any additional changes.

Does Boa need adding to the readme if it’s supported  in https://github.com/CanadaHonk/eshost ?

**Comment by WorrierKhushal** [comment]
@jasonwilliams 
Hi, I’m interested in working on this issue.

Is this still open for contribution? 
I’d like to start by understanding the current status and testing setup.

Please let me know if I can proceed.

**Comment by ptomato** [comment]
Also relevant
- https://github.com/devsnek/esvu/pull/71
- https://github.com/devsnek/esvu/pull/66
- https://github.com/tc39/eshost/pull/147

**Comment by ParthMozarkar** [comment]
Hi! I’m interested in contributing to Boa and noticed this tracking issue for `JsTypedArray` methods.

If there are still missing or partially implemented methods related to `JsTypedArray`, I’d be happy to help work on implementing or improving them. Please let me know if there is a specific method or task from this tracking issue that would be a good place to start.

Thanks!

**Comment by 0xShubhamSolanki** [comment]
Hi @jasonwilliams,

I was going through this issue and the related discussion about adding Boa support to the Test262 harness.

From what I understand, Boa already seems to be supported in the eshost fork maintained by @CanadaHonk, and the main question now is whether anything else is needed for proper Test262 integration or documentation.
I'd like to help with this if it's still open. I can start by checking the current setup with the CanadaHonk forks and see if any integration or documentation updates are needed.

Please let me know if that sounds like a good direction. Thanks!

**Comment by SKM2227229725** [comment]
Hi! I’m interested in working on this issue.

From the discussion it seems Boa support already exists in the eshost fork maintained by @CanadaHonk.

Would the next step be updating Boa’s documentation and testing setup to use that fork, or is additional integration still required?

**Comment by ivankra** [comment]
This should be the next step: https://github.com/boa-dev/boa/issues/5054

**Comment by ParthMozarkar** [comment]
could you assign this to me?

**Comment by ParthMozarkar** [comment]
> could you assign this to me?

could this issue be assigned to me?

**Comment by ivankra** [comment]
> Boa support has now been added to esvu, jsvu, and eshost.

No, you just sent some PRs that aren't getting merged any time soon unless you chase maintainers and convince them to merge your work. Have you run the whole thing end-to-end and validated against `boa_tester`?

This PR will be needed to fix some of the discrepancies - https://github.com/boa-dev/boa/pull/5234
